pkg/operator/ceph/file/filesystem.go

```diff
@@ -249,3 +271,18 @@ func downFilesystem(context *clusterd.Context, clusterName, filesystemName strin
 	logger.Infof("Downed filesystem %s", filesystemName)
 	return nil
 }
+
+// GenerateMetadataPoolName is a helper function that generates a new name for the given metadata pool and returns the renamed pool.
+func GenerateMetadataPoolName(fsName string, metadataPool *model.Pool) *model.Pool {
```

Comment:
Return a string instead of updating the `metadataPool.Name`. The caller should decide if they want to update the struct.

Reply:
I have done that, please check.
Thanks